modules/virtualization/microvm/appvm.nix

```diff
@@ -0,0 +1,176 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  configHost = config;
+  cfg = config.ghaf.virtualization.microvm.appvm;
+  waypipe-ssh = pkgs.callPackage ../../../user-apps/waypipe-ssh {};
+
+  makeVm = {vm}: let
+    hostname = "vm-" + vm.name;
+    appvmConfiguration = {
+      imports = [
+        ({
+          lib,
+          config,
+          ...
+        }: {
+          ghaf = {
+            users.accounts.enable = lib.mkDefault configHost.ghaf.users.accounts.enable;
```

Comment:
Should we actually not share the `configHost.ghaf.users.accounts` enabling with the appvms but have enabling appVMs accounts separately from host? 

I know for `-debug` targets it makes development easier but the problem is that `-release` target should not allow interactive login - like `getty` - to host at all. In that scenario, the host systemd services are run with unprivileged user (similar to `ghaf`) but we likely do not want the share the accounts declarations between the host and the appVMS by default. In some special, unforseen scenario(s) sharing the host and appVMs user account(s) might be needed but that should be exception.